tests/TransactionBuilderTest.php

```diff
@@ -0,0 +1,220 @@
+        DB::commit();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'] + 1, $user['age']);
+    }
+
+    public function testDecrement()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->decrement('age');
+        DB::rollBack();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'], $user['age']);
+
+        /** commit test */
+        DB::beginTransaction();
+        DB::collection('users')->where('name', $this->originData['name'])->decrement('age');
+        DB::commit();
+        $user = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($this->originData['age'] - 1, $user['age']);
+    }
+
+    public function testQuery()
+    {
+        /** rollback test */
+        DB::beginTransaction();
+        $count = DB::collection('users')->count();
+        $this->assertEquals(1, $count);
+        DB::collection('users')->insert($this->insertData);
+        $count = DB::collection('users')->count();
+        $this->assertEquals(2, $count);
+        DB::rollBack();
+        $count = DB::collection('users')->count();
+        $this->assertEquals(1, $count);
+
+        /** commit test */
+        DB::beginTransaction();
+        $count = DB::collection('users')->count();
+        $this->assertEquals(1, $count);
+        DB::collection('users')->insert($this->insertData);
+        $count = DB::collection('users')->count();
+        $this->assertEquals(2, $count);
+        DB::commit();
+        $count = DB::collection('users')->count();
+        $this->assertEquals(2, $count);
+    }
+
+    public function testTransaction()
+    {
+        $count = DB::collection('users')->count();
+        $this->assertEquals(1, $count);
+
+        $new_age = $this->originData['age'] + 1;
+        DB::transaction(function () use ($new_age) {
+            DB::collection('users')->insert($this->insertData);
+            DB::collection('users')->where('name', $this->originData['name'])->update(['age' => $new_age]);
+        });
+        $count = DB::collection('users')->count();
+        $this->assertEquals(2, $count);
+
+        $checkInsert = DB::collection('users')->where('name', $this->insertData['name'])->first();
+        $this->assertNotNull($checkInsert);
+        $this->assertEquals($this->insertData['age'], $checkInsert['age']);
+
+        $checkUpdate = DB::collection('users')->where('name', $this->originData['name'])->first();
+        $this->assertEquals($new_age, $checkUpdate['age']);
+    }
+
+    /**
+     * Supports infinite-level nested transactions, but outside transaction rollbacks do not affect the commit of inside transactions
+     */
+    public function TestNestingTransaction()
```

Comment:
Method should start with a lowercase "t" for consistency.

Reply:
Yes I noted it too in my comment [Comment](https://github.com/levon80999/laravel-mongodb/pull/3#discussion_r994895176)